lib/routes/linkresearcher/index.ts

```diff
@@ -1,76 +1,117 @@
+    const token = crypto.randomUUID();
 
-    let data = { filters: { status: false } };
-    if (query.subject !== undefined && query.columns !== undefined) {
-        data = { filters: { status: true, subject: query.subject, columns: query.columns } };
-        title = `${title}「${query.subject} & ${query.columns}」`;
-    } else if (query.subject !== undefined && query.columns === undefined) {
-        data = { filters: { status: true, subject: query.subject } };
-        title = `${title}「${query.subject}」`;
-    } else if (query.subject === undefined && query.columns !== undefined) {
-        data = { filters: { status: true, columns: query.columns } };
-        title = `${title}「${query.columns}」`;
+    const data: {
+        filters: {
+            status: boolean;
+            subject?: string;
+            columns?: string;
+        };
+    } = { filters: { status: true } };
+
+    if (subject) {
+        data.filters.subject = subject;
+        title = `${title}「${subject}」`;
     }
-    data.query = query.query;
+
+    if (columns) {
+        data.filters.columns = columns;
+        title = `${title}「${columns}」`;
+    }
+
     const dataURL = `${baseURL}/api/${category === 'careers' ? 'articles' : category}/search`;
-    const pageResponse = await got.post(dataURL, {
+    const pageResponse = await ofetch<{
+        hits: SearchResultItem[];
+    }>(dataURL, {
+        method: 'POST',
         headers: {
             'content-type': 'application/json; charset=UTF-8',
-            'x-xsrf-token': xsrfToken,
-            cookie: `XSRF-TOKEN=${xsrfToken}`,
+            'x-xsrf-token': token,
+            cookie: `XSRF-TOKEN=${token}`,
         },
-        searchParams: {
+        params: {
             from: 0,
             size: 20,
             type: category === 'careers' ? 'CAREER' : 'SEARCH',
         },
-        json: data,
+        body: {
+            ...data,
+            query,
+        },
     });
 
-    const list = pageResponse.data.hits;
-
-    const out = list.map((item) => ({
-        title: item.title,
-        description: item.content,
-        pubDate: parseDate(item.createdAt, 'x'),
-        link: `${metaURL}/${item.id}`,
-        guid: `${metaURL}/${item.id}`,
-        doi: item.identCode === undefined ? '' : item.identCode,
-        author: item.authors === undefined ? '' : item.authors.join(', '),
-    }));
+    const items = await Promise.all(
+        pageResponse.hits.map((item) => {
+            const link = `${baseURL}/${category}/${item.id}`;
+            return cache.tryGet(link, async () => {
+                const response = await ofetch<DetailResponse>(`${apiURL}/${category === 'theses' ? 'theses' : 'information'}/${item.id}`, {
+                    responseType: 'json',
+                });
+                const dataItem: DataItem = {
+                    title: response.title,
+                    description: response.content,
```

Comment:
Some theses have `enTextList` and `zhTextList` and it's suggested to use them over the `content` only

Reply:
You are right. I'll fix it.